.azurepipelines/MuDevOpsWrapper.yml

```diff
@@ -0,0 +1,88 @@
+## @file
+# Azure Pipeline build file for a build using mu_devops.
+#
+# NOTE: This file is automatically synchronized from Mu DevOps. Update the original file there
+#       instead of the file in this repo.
+#
+# - Mu DevOps Repo: https://github.com/microsoft/mu_devops
+# - File Sync Settings: https://github.com/microsoft/mu_devops/blob/main/.sync/Files.yml
+#
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+resources:
+  repositories:
+    - repository: mu_devops
+      type: github
+      endpoint: microsoft
+      name: microsoft/mu_devops
+      ref: refs/tags/v4.0.0
+
+parameters:
+- name: do_ci_build
+  displayName: Perform Stuart CI Build
+  type: boolean
+  default: true
+- name: do_ci_setup
+  displayName: Perform Stuart CI Setup
+  type: boolean
+  default: true
+- name: do_non_ci_build
+  displayName: Perform non-CI Stuart Build
+  type: boolean
+  default: false
+- name: do_non_ci_setup
+  displayName: Perform non-CI Stuart Setup
+  type: boolean
+  default: false
+- name: do_pr_eval
+  displayName: Perform Stuart PR Evaluation
+  type: boolean
+  default: true
+- name: container_build
+  displayName: Flag for whether this repo should do stuart_setup
+  type: boolean
+  default: false
+- name: os_type
+  displayName: OS type on the self-hosted agent pools
+  type: string
+  values:
+  - Windows_NT
+  - Linux
+  default: Windows_NT
+- name: build_matrix
+  displayName: Build matrix for this repository
+  type: object
+- name: pool_name
+  displayName: Variable name that hosts pool name to be used for self-hosted agents
+  type: string
+  default: pool_name

```

Comment:
Is the default really meant to be `"pool_name"`? Checking because typically the parameter would be defined/overridden with a pipeline value or be empty as a default.

Reply:
Yeah, I think we do not have to set this default to `pool_name`. This was set to `pool_name` in synced file so that the applicable repos do not have to supply the variable of `pool_name` in their top level yaml across all repos.

Another catch is that this is actually the "variable name" instead of the "variable value". :) So that it is indeed driven by a variable named "pool_name". This was done because the pool name _value_ will be used in 2 different scenarios: one is in the condition, so that we can skip the jobs that indicates selfhost run but does not have selfhost pool specified; the other is to specify the pool name. In order to resolve the variables properly during the pipeline run, we set it this way so that the variables have the correct value when it is being evaluated.

Reply:
Resolving this for now given 1. the consumer using the Jobs/PRGate can consume the default empty parameter, and the sycned repos are in a more controlled environment which could use the "pool_name" as default to simplify the top level scripts. We can revisit this if deem needed later.